packages/api-server/api_server/rmf_io/transport.py

```diff
@@ -0,0 +1,43 @@
+from typing import Optional
+
+import rclpy
+from building_map_msgs.msg import BuildingMap
+from rclpy.node import Node as RosNode
+from rclpy.subscription import Subscription
+from rmf_door_msgs.msg import DoorState
+
+from .rmf_io import RmfGateway
+
+
+class RmfTransport:
+    def __init__(self, ros2_node: RosNode, rmf_gateway: RmfGateway):
+        self.ros2_node = ros2_node
+        self.rmf_gateway = rmf_gateway
+        self.door_states_sub: Optional[Subscription] = None
+        self.building_map_sub: Optional[Subscription] = None
+
+    def subscribe_all(self):
+        self.door_states_sub = self.ros2_node.create_subscription(
+            DoorState, "door_states", self.rmf_gateway.door_states.on_next, 10
+        )
+
+        self.building_map_sub = self.ros2_node.create_subscription(
+            BuildingMap,
+            "map",
+            self.rmf_gateway.building_map.on_next,
+            rclpy.qos.QoSProfile(
+                history=rclpy.qos.HistoryPolicy.KEEP_LAST,
```

Comment:
Dump question, Should we use the `KEEP_LAST` since we already have a database that saves the states? Or it's better to have it both?

Reply:
rmf only publishes the building map once, if we don't use `KEEP_LAST`, we must start the server before rmf else we will never receive any building map.

Reply:
Cool, thanks for the feedback.